src/layouts/ReviewRequest/Dashboard.tsx

```diff
@@ -76,19 +76,22 @@ export const ReviewRequestDashboard = (): JSX.Element => {
     // TODO!
     isError: isMergeError,
   } = useMergeReviewRequest(siteName, prNumber, false)
+  const [isApproved, setIsApproved] = useState<boolean | null>(null)
 
   const { onCopy, hasCopied } = useClipboard(data?.reviewUrl || "")
-
   const reviewStatus = data?.status
-  const isApproved = reviewStatus === ReviewRequestStatus.APPROVED
+
   useEffect(() => {
     if (
       reviewStatus === ReviewRequestStatus.CLOSED ||
       reviewStatus === ReviewRequestStatus.MERGED
     ) {
       setRedirectToPage(`/sites/${siteName}/dashboard`)
     }
-  }, [reviewStatus, setRedirectToPage, siteName])
+    if (reviewStatus && isApproved === null) {
```

Comment:
hey I am confused with this code here :( 
why are we setting it to approved when `isApproved` is null? if the idea to set them when we first run this code, this wouldn't play well well with a flow like `approve` -> `disapprove` -> `approve`? shouldnt this then just be ```if(reviewStatus) setIsApproved(reviewStatus === ReviewRequestStatus.APPROVED)```?

Reply:
The idea for this code is to set `isApproved` only on initial load - `isApproved` is set to null originally, and if `reviewStatus` has finished loading, we set `setIsApproved(reviewStatus === ReviewRequestStatus.APPROVED)`. From this point onwards, isApproved is only affected by the user's changes, which are immediate (as opposed to the previous behaviour of having to wait for the query from the backend)

Without the check for the first load, it gets stuck on the initial state of the page and has to rely on new information from the backend in order to change state!